plugins/variable-import-plugins/complex-variable-flattener/README.md

```diff
@@ -0,0 +1,214 @@
+   "lastName": "Doe"
+  }
+  ```
+  and want to analyse these variable in Optimize as `user.firstName` & `user.lastName`.
+
+### How does it work?
+
+To see how the import in Optimize works in general, have a look at the [documentation][2].
+The details of the variable import and how the plugin hooks into that
+are explained in the [variable import plugin readme][3].
+
+### Setup your environment
+
+First, add the Optimize plugin dependency to your project using mavens [pom.xml][4]:
+
+```xml
+<dependency>
+  <groupId>org.camunda.optimize</groupId>
+  <artifactId>plugin</artifactId>
+  <version>${optimize.version}</version>
+</dependency>
+```
+
+To tell maven where to find the plugin environment, add the following repository to your project:
+
+```xml
+<repositories>
+  <repository>
+    <id>camunda-bpm-nexus</id>
+    <name>camunda-bpm-nexus</name>
+    <url>
+      https://app.camunda.com/nexus/content/repositories/camunda-optimize
+    </url>
+  </repository>
+</repositories>
+```
+
+Then you also need to create an uber jar, which can be done with the following:
+```xml
+  <build>
+    <defaultGoal>install</defaultGoal>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-assembly-plugin</artifactId>
+        <version>3.1.0</version>
+        <executions>
+          <execution>
+            <phase>package</phase>
+            <goals>
+              <goal>single</goal>
+            </goals>
+            <configuration>
+              <finalName>${project.artifactId}</finalName>
+              <descriptorRefs>
+                <descriptorRef>jar-with-dependencies</descriptorRef>
+              </descriptorRefs>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <artifactId>maven-jar-plugin</artifactId>
+        <version>3.1.2</version>
+        <executions>
+          <execution>
+            <id>default-jar</id>
+            <phase>none</phase>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+```
+Note: For further information why it is necessary to build the plugin as an uber jar you can have a look
+at [plugin setup][4] in the Optimize documentation.
+
+### Implement the variable plugin
+
+In this example we assume that you have a business process where you interact with
+users, e.g. handling customer support. Each user is stored as a JSON variable
+during the process execution. Now you want to analyze how much time you need for the
+support for a certain user and how you can improve the performance of the
+customer support process, which Optimize would be the perfect tool for. Since the
+user data is stored in a complex JSON variable, Optimize is not able to analyze
+it out of the box. However, we can use the Optimize plugin system to hook into the
+import and transform each complex user variable to a set of primitive variables.
+
+For that extend the
```

Comment:
```suggestion
To do this, you need to extend the
```

Reply:
:+1: will fix that in other readmes as well!